src/invocation/ConnectionAuthenticator.ts

```diff
@@ -22,42 +22,60 @@ import {ClientConnection} from './ClientConnection';
 
     private connection: ClientConnection;
     private client: HazelcastClient;
     private clusterService: ClusterService;
+    private logger = LoggingService.getLoggingService();
 
     constructor(connection: ClientConnection, client: HazelcastClient) {
         this.connection = connection;
         this.client = client;
         this.clusterService = this.client.getClusterService();
     }
 
-    authenticate(ownerConnection: boolean): Promise<void> {
-        var credentials: ClientMessage = this.createCredentials(ownerConnection);
-
+    authenticate(asOwner: boolean): Promise<void> {
+        var credentials: ClientMessage = this.createCredentials(asOwner);
         return this.client.getInvocationService()
             .invokeOnConnection(this.connection, credentials)
             .then((msg: ClientMessage) => {
                 var authResponse = ClientAuthenticationCodec.decodeResponse(msg);
-                if (authResponse.status === 0) {
-                    this.connection.setAddress(authResponse.address);
-                    this.connection.setConnectedServerVersion(authResponse.serverHazelcastVersion);
-                    if (ownerConnection) {
-                        this.clusterService.uuid = authResponse.uuid;
-                        this.clusterService.ownerUuid = authResponse.ownerUuid;
-
-                    }
-                } else {
-                    throw new Error('Could not authenticate connection to ' + this.connection.getAddress().toString());
+                switch (authResponse.status) {
+                    case AuthenticationStatus.AUTHENTICATED:
+                        this.connection.setAddress(authResponse.address);
+                        this.connection.setConnectedServerVersion(authResponse.serverHazelcastVersion);
+                        if (asOwner) {
+                            this.clusterService.uuid = authResponse.uuid;
+                            this.clusterService.ownerUuid = authResponse.ownerUuid;
+
+                        }
+                        this.logger.debug('ConnectionAuthenticator',
+                            'Connection to ' +
+                            this.connection.getAddress().toString() + ' authenticated');
+                        break;
+                    case AuthenticationStatus.CREDENTIALS_FAILED:
+                        this.logger.error('ConnectionAuthenticator', 'Invalid Credentials' );
+                        throw new Error('Invalid Credentials, could not authenticate connection to ' +
+                            this.connection.getAddress().toString());
+                    case AuthenticationStatus.SERIALIZATION_VERSION_MISMATCH:
+                        this.logger.error('ConnectionAuthenticator', 'Serialization version mismatch' );
+                        throw new Error('Serialization version mismatch, could not authenticate connection to ' +
+                            this.connection.getAddress().toString());
```

Comment:
what if the connection fails with an unknown authentication status? A new authentication status may be added in the future. This is important for forward-compatibility.
Also can you add a test for this scenario?